api/src/main/java/com/example/api/service/validator/ChapterValidator.java

```diff
@@ -36,4 +37,19 @@ public void validateChapterCreation(ChapterForm form) throws RequestValidationEx
             throw new RequestValidationException(ExceptionMessage.TWO_CHAPTERS_ON_THE_SAME_POSITION);
         }
     }
+
+    public void validateChapterEdition(ChapterForm form, Chapter chapter) throws RequestValidationException{
+        ActivityMap chapterMap = chapter.getActivityMap();
+
+        Integer newSizeX = form.getSizeX();
+        Integer newSizeY = form.getSizeY();
+
+        if (chapterMap.getSurveys().stream().anyMatch(survey -> survey.getPosX() >= newSizeX || survey.getPosY() >= newSizeY)
+                || chapterMap.getFileTasks().stream().anyMatch(fileTask -> fileTask.getPosX() >= newSizeX || fileTask.getPosY() >= newSizeY)
+                || chapterMap.getGraphTasks().stream().anyMatch(graphTask -> graphTask.getPosX() >= newSizeX || graphTask.getPosY() >= newSizeY)
+                || chapterMap.getInfos().stream().anyMatch(info -> info.getPosY() >= newSizeY || info.getPosX() >= newSizeX)){
```

Comment:
Można to chyba trochę uprościć
<img width="970" alt="Zrzut ekranu 2022-09-29 o 19 42 51" src="https://user-images.githubusercontent.com/56938330/193104253-629fcaa3-9f7b-4915-8ec4-c6672051ddae.png">